src/Workspaces/Core/Portable/Workspace/Host/TaskScheduler/TaskSchedulerProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.Host.Mef;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    [ExportWorkspaceService(typeof(ITaskSchedulerProvider), ServiceLayer.Default)]
+    [Shared]
+    internal sealed class TaskSchedulerProvider : ITaskSchedulerProvider
+    {
+        [ImportingConstructor]
```

Comment:
isn't this supposed to have some Obsolete attribute on it now to prevent direct instantiation?

Reply:
I guess it could have `[Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]`, but we don't have it on all importing constructors.
I wonder why do we need to add more clutter (extra Obsolete attribute) if we can have an analyzer that checks for the presence of ImportingConstructor at the constructor call site.

Reply:
tagging @sharwell .  My understanding of `Obsolete` is that it's the most convenient way to prevent people for directly instantiating the type.

Reply:
Added Obsolete.

Reply:
@tmat I need to enable the analyzer and clean up all the existing cases.